Managers/WindowMan.cpp

```diff
@@ -0,0 +1,602 @@
+
+			if (displayIndex == m_DisplayIndexPrimaryWindowIsAt) {
+				m_MultiDisplayWindows.emplace_back(m_PrimaryWindow);
+				m_MultiDisplayRenderers.emplace_back(m_PrimaryRenderer);
+			} else {
+				m_MultiDisplayWindows.emplace_back(SDL_CreateWindow(nullptr, displayOffsetX, displayOffsetY, displayWidth, displayHeight, SDL_WINDOW_FULLSCREEN_DESKTOP | SDL_WINDOW_SKIP_TASKBAR), SDLWindowDeleter());
+				if (m_MultiDisplayWindows.back()) {
+					m_MultiDisplayRenderers.emplace_back(SDL_CreateRenderer(m_MultiDisplayWindows.back().get(), -1, SDL_RENDERER_ACCELERATED), SDLRendererDeleter());
+					if (!m_MultiDisplayRenderers.back()) {
+						errorSettingFullscreen = true;
+					}
+				} else {
+					errorSettingFullscreen = true;
+				}
+				if (errorSettingFullscreen) {
+					break;
+				}
+			}
+
+			int textureOffsetX = (displayOffsetX - m_DisplayArrangementLeftMostOffset);
+			int textureOffsetY = (displayOffsetY - m_DisplayArrangementTopMostOffset);
+
+			m_MultiDisplayTextureOffsets.emplace_back(SDL_Rect{
+				textureOffsetX / resMultiplier,
+				// Sometimes an odd Y offset implodes all the things, depending on the stupidity of the arrangement and what display is primary.
+				// Sometimes it needs to be in multiples of 4 for reasons unknown to man, so we're just gonna go with this and hope for the best, for now at least.
+				RoundToNearestMultiple(textureOffsetY, 2) / resMultiplier,
+				displayWidth / resMultiplier,
+				displayHeight / resMultiplier
+			});
+		}
+
+		if (errorSettingFullscreen) {
+			ClearMultiDisplayData();
+			SDL_SetWindowPosition(m_PrimaryWindow.get(), windowPrevPositionX, windowPrevPositionY);
+			return false;
+		}
+
+		CreateMultiDisplayTextures();
+		SDL_SetWindowFullscreen(m_PrimaryWindow.get(), SDL_WINDOW_FULLSCREEN_DESKTOP);
+
+		return true;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void WindowMan::DisplaySwitchIn() const {
+		g_UInputMan.DisableMouseMoving(false);
+		if (!m_MultiDisplayWindows.empty()) {
+			SDL_RaiseWindow(m_PrimaryWindow.get());
+			for (auto &window : m_MultiDisplayWindows) {
+				SDL_RaiseWindow(window.get());
+			}
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void WindowMan::DisplaySwitchOut() const {
+		g_UInputMan.DisableMouseMoving(true);
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void WindowMan::HandleWindowEvent(const SDL_Event &windowEvent) {
+		switch (windowEvent.window.event) {
+			case SDL_WINDOWEVENT_FOCUS_GAINED:
+				if (!m_FocusLostDueToMovingBetweenGameWindows) {
+					DisplaySwitchIn();
+				}
+				m_AnyWindowHasFocus = true;
+				break;
+			case SDL_WINDOWEVENT_FOCUS_LOST:
+				m_AnyWindowHasFocus = false;
+				m_FocusLostDueToMovingBetweenGameWindows = true;
```

Comment:
Does this event not trigger when the game alt-tabs or whatever? Cause it seems weird that this flag is set even if the game isn't in multi display mode.